src/manager.cpp

```diff
@@ -151,13 +166,107 @@ void Manager::deleteSingleFruVpd(
     logging::logMessage(std::string(i_dbusObjPath));
 }
 
-std::string Manager::getExpandedLocationCode(
-    const sdbusplus::message::object_path& i_dbusObjPath)
+std::string Manager::getExpandedLocationCode(const std::string& i_locationCode,
+                                             uint16_t i_nodeNumber)
 {
-    // Dummy code to supress unused variable warning. To be removed.
-    logging::logMessage(std::string(i_dbusObjPath));
+    (void)i_nodeNumber;
+    std::string o_expandedLocationCode;
+    if (!isValidLocationCode(i_locationCode))
+    {
+        throw types::CommonError::InvalidArgument();
+    }
 
-    return std::string{};
+    nlohmann::json l_parsedJson = m_worker->getJsonObject();
+    if (!l_parsedJson.contains("frus"))
+    {
+        throw JsonException("Missing frus section in JSON", l_parsedJson);
+    }
+
+    const nlohmann::json& l_listOfFrus =
+        l_parsedJson["frus"].get_ref<const nlohmann::json::object_t&>();
+
+    try
+    {
+        for (const auto& l_frus : l_listOfFrus.items())
+        {
+            for (const auto& l_inventoryItem : l_frus.value())
+            {
+                if (l_inventoryItem["extraInterfaces"].find(
```

Comment:
Use "contains" instead of find.  Will make your code look clean also you will not not have to compare to the end.

Reply:
Updated